src/app/common/FabricTableImpl.h

```diff
@@ -0,0 +1,143 @@
+/**
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <lib/support/TypeTraits.h>
+
+namespace chip {
+namespace app {
+namespace common {
```

Comment:
We're really bad at namespace consistency, but most of the namespaces other than "chip" and "app" use PascalCase-style naming...

Separately, though, "common" is in this case unfortunately "generated-ish stuff common to all apps" or something.  Check with @andy31415 about what it's supposed to be....

So practically, I would suggest a different subdirectory for these table bits.  Which immediately raises the question of whether this should be under app at all or under lib/support....

Reply:
Updated names respectively to PascalCase, and as per @andy31415 , changed to `src/app/storage` to be more specific vs overly-generic `common` - but he agreed `src/app` seemed like a good place as this is meant for servers